sputnikdao2/src/lib.rs

```diff
@@ -0,0 +1,312 @@
+use near_contract_standards::fungible_token::metadata::{
+    FungibleTokenMetadata, FungibleTokenMetadataProvider, FT_METADATA_SPEC,
+};
+use near_contract_standards::fungible_token::FungibleToken;
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::collections::LookupMap;
+#[cfg(target_arch = "wasm32")]
+use near_sdk::env::BLOCKCHAIN_INTERFACE;
+use near_sdk::json_types::{Base64VecU8, ValidAccountId, U128};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{env, near_bindgen, AccountId, Balance, PanicOnDefault, Promise, PromiseOrValue};
+
+use crate::bounties::{Bounty, BountyClaim};
+pub use crate::policy::{Policy, RoleKind, VersionedPolicy};
+pub use crate::proposals::{Proposal, ProposalInput, ProposalKind, ProposalStatus};
+pub use crate::types::{Action, Config};
+
+mod bounties;
+mod policy;
+mod proposals;
+mod types;
+pub mod views;
+
+near_sdk::setup_alloc!();
+
+#[cfg(target_arch = "wasm32")]
+const BLOCKCHAIN_INTERFACE_NOT_SET_ERR: &str = "Blockchain interface not set.";
+
+/// Container for all contract data.
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct ContractData {
+    /// DAO configuration.
+    pub config: Config,
```

Comment:
You may want to wrap config into `LazyOption` to avoid deserializing it every time. And then extract `bond` from `config` and store it separately (as a copy). Otherwise, FTMetadata fields such as icon can be pretty expensive to deserialize, imaging a 500 bytes icon, and also description + purpose
```suggestion
    pub config: LazyOption<Config>,
```